libs/utils/android/screen.py

```diff
@@ -136,4 +137,25 @@ def get_screen_density(target):
         """
         return target.execute('getprop ro.sf.lcd_density')
 
+    @staticmethod
+    def is_screen_on(target):
+	cmdout = target.execute('dumpsys input_method')
+	return ('mInteractive=true' in cmdout)
+
+    @staticmethod
+    def set_screen_on(target):
```

Comment:
Moreover, I've found these two guys:
https://developer.android.com/reference/android/view/KeyEvent.html#KEYCODE_WAKEUP
https://developer.android.com/reference/android/view/KeyEvent.html#KEYCODE_SLEEP

Thus perhaps we can avoid the "expensive" check with dumpsys and go straight with these key events depending on the value of the `on` parameter.